code/network/lib/Loot/Network/ZMQ/Server.hs

```diff
@@ -173,7 +171,11 @@ runBroker = do
             Publish k v -> publish k v
 
     let frontToListener = \case
-            (cId:"":msgT:msg) -> do
+            [cId,"","getid"] -> do
+                Z.sendMulti ztServFront $
+                  NE.fromList [cId,"",unZTInternalId ztOurNodeId]
+                ztLog ztServLogging Debug "Received request connection, replied with our id"
+            (cId:"":"m":msgT:msg) -> do
                 ztEnv <- atomically $ runMaybeT $ do
                     lId <- MaybeT $ Map.lookup (MsgType msgT) <$> readTVar ztMsgTypes
                     MaybeT $ Map.lookup lId <$> readTVar ztListeners
```

Comment:
I can understand what is going on here, if I concentrate on those words between `[run]MaybeT`, but its hard. Can you help, please?

Reply:
Yes sure. 
L180: get the listener id that uses this message type.
L181: resolve these listener id into the listener

MaybeT will fail if any of these will fail.